### PR TITLE
Update question 206

### LINKED ISSUE #127: Update question 206
### Thanks for helping!

Thank you for helping to port [this question](../blob/master/questions/206/) from C++17 to C++23.

You'll find the source code in [question.cpp](../blob/master/questions/206/question.cpp), this should normally not need modification.

In C++17, the correct answer is:
> The program is guaranteed to output: `1`

Please verify that this is still correct in C++23. If it is, please do the following:
- [ ] Update [explanation.md](../blob/master/questions/206/explanation.md):
  - [ ] Refer to the correct section numbers
  - [ ] Use updated quotes from those sections (the wording might have changed)
  - [ ] Make sure the rest of the text in the explanation is consistent with the new standard
- [ ] Update [hint.md](../blob/master/questions/206/hint.md) if needed (usually not needed)

*Note that some older questions are missing references to the standard, we don't need to fix that now.*

If you get stuck, please describe the problem in detail in a comment to the issue, and a maintainer will add the `help wanted` tag so others can chime in.

If the correct answer has changed from C++17 to C++23, you can either just leave a comment in this issue, or see the instructions for [updating meta data](../blob/master/METADATA_HOWTO.md).

### PR DESCRIPTION
**Pull request by ani-hovhannisyan**
Fixes #127.

### CODE REVIEW

**Review by tocic — CHANGES_REQUESTED** [review]
I think there are issues with your references. For example, the first one is expected to be rendered as the link
https://timsong-cpp.github.io/cppwp/std23/expr.unary#8.3. But it doesn't lead to the expected grammar clause for `sizeof` (consider using §[expr.unary.general]¶1 instead).

**Comment by ani-hovhannisyan** [comment]
Yes, I wasn't sure about the links as I was looking at this works.https://timsong-cpp.github.io/cppwp/n4659/expr.unary.
Thanks for the correction. Please check if the change fixed links and §[expr.sizeof]¶1  explanation.

File: questions/206/explanation.md [hunk 1]
```diff
@@ -11,10 +11,10 @@ The grammar for `sizeof` is in §[expr.unary]¶1:
...
 
-We can see that `(0)["abcdefghij"]` is identical to `("abcdefghij")[0]` from §[expr.sub]¶1 which says:
+We can see that `(0)["abcdefghij"]` is identical to `("abcdefghij")[0]` from §[expr.sub]¶2 which says:
 
 > ... The expression `E1[E2]` is identical (by definition) to `*((E1)+(E2))` ...
 
 So we end up with 0th element of `"abcdefghij"`, which is `a`, which is a `char`. And the result of `sizeof('a')` will be `1` since §[expr.sizeof]¶1 says:
 
-> ... `sizeof(char)`, `sizeof(signed char)` and `sizeof(unsigned char)` are `1` ...
+> The result of sizeof applied to any of the narrow character types is 1.
```
Comment (tocic) [inline questions/206/explanation.md:20]: Maybe it's also worth explaining that `char` is a narrow character type like [here](https://github.com/knatten/cppquiz23/pull/218).

**Review by tocic — APPROVED** [review]

File: questions/206/explanation.md (hunk 1 above)
Comment (knatten) [reply questions/206/explanation.md:20]: Yeah I think that would be good.

**Comment by ani-hovhannisyan** [comment]
Sorry for the inconvenience. Please feel free to remove this PR. I will submit a new PR with separate branches for each question update, taking your comments into account.

**Pull request closed without merging**